libtransmission/torrent-magnet.cc

```diff
@@ -304,9 +304,9 @@ static void onHaveAllMetainfo(tr_torrent* tor, tr_incomplete_metadata* m)
         tor->incompleteMetadata = nullptr;
         tor->isStopping = true;
         tor->magnetVerify = true;
-        if (tor->session->shouldPauseAddedTorrents())
+        if (!tor->session->shouldPauseAddedTorrents() && tor->startAfterVerify)
         {
-            tor->startAfterVerify = false;
+            tor->startAfterVerify = true;
```

Comment:
I think this revision just reverses the previous one's mistake: now we only set `startAfterVerify` to true iff it's already true :smile_cat: 

I think the root cause is with how confusing the start / stop / verify state variables are and that they need a rethink. Several people have taken a run at this code and had trouble with it :thought_balloon:

Reply:
yeah, I don't know how other clients work. but in the macOS client, you can override the global start option in a modal. And when you do use the override, it sets `startAfterVerify` to true - which is why im also checking for that
<img width="526" alt="Screenshot 2023-01-04 at 3 44 44 pm" src="https://user-images.githubusercontent.com/7323792/210486186-db94c6f5-625a-448b-bddd-9c4cd7710631.png">